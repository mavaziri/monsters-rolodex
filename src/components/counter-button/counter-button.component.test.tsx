import { fireEvent, render, screen } from '@testing-library/react';

import CounterButton from './counter-button.component';

describe('CounterButton Component', () => {
  test('expect to render CounterButton component', () => {
    const { container } = render(<CounterButton />);
    expect(container).toMatchSnapshot();
  });

  test('correctly increment the counter', () => {
    render(<CounterButton />);

    // Get the button and counter display
    const button = screen.getByRole('button', { name: /increment/i });
    const counterText = screen.getByText(/count/i);

    // Initial state
    expect(counterText).toHaveTextContent('Count: 0');

    // Simulate a click on the button
    fireEvent.click(button);

    // Verify that the count has incremented
    expect(counterText).toHaveTextContent('Count: 1');
  });
});
