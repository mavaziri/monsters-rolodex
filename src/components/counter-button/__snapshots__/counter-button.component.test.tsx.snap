// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterButton Component expect to render CounterButton component 1`] = `
<div>
  <div>
    <p>
      Count: 
      0
    </p>
    <button>
      Increment
    </button>
  </div>
</div>
`;

exports[`expect to render CounterButton component 1`] = `
<div>
  <div>
    <p>
      Count: 
      0
    </p>
    <button>
      Increment
    </button>
  </div>
</div>
`;
